tests/app/main/views/test_index.py

```diff
@@ -43,6 +43,17 @@ def it_should_show_future_events_in_cards(self, client, sample_future_events, sa
 
         assert content == intro_course['title']
 
+    def it_should_show_past_and_future_events_in_cards(self, client, sample_articles_summary, sample_all_events):
+        response = client.get(url_for(
+            'main.index'
+        ))
+        page = BeautifulSoup(response.data.decode('utf-8'), 'html.parser')
+        content = page.find("div", {"class": "past_corner"}).string
+
+        past_event = [e for e in sample_all_events if e['past'] == 'Past Event'][0]
```

Comment:
How comes when i try and find the tag ['past'] i get a ` KeyError: 'past'`. But if i change the key to another of the other tags in the mockup it gets the information fine. Any ideas?

Reply:
Not all elements have the `past` flag set, I would use `e.get('past')` instead as this will return `None` if it can't find the key